### PR TITLE
Display survey prompt once per session

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #4077

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/dataScienceSurveyBanner.ts [hunk 1]
```diff
@@ -140,7 +140,8 @@ export class DataScienceSurveyBanner implements IJupyterExtensionBanner {
         if (!show) {
             return;
         }
-
+        // Disable for the current session.
+        this.disabledInCurrentSession = true;
```
Comment (IanMatthewHuff) [inline src/client/datascience/dataScienceSurveyBanner.ts:144]: Is this the right location? It's still after an await statement, and also after the disabledInCurrentSession check. So can't two calls to show banner both get released?

File: src/client/datascience/insidersNativeNotebookSurveyBanner.ts [hunk 2]
```diff
@@ -106,8 +107,7 @@ export class InsidersNativeNotebooksSurveyBanner implements IJupyterExtensionBan
                 break;
             }
             default: {
-                // Disable for the current session.
-                this.disabledInCurrentSession = true;
+                //
```
Comment (IanMatthewHuff) [inline src/client/datascience/insidersNativeNotebookSurveyBanner.ts:110]: Extra comment marker?

File: src/client/datascience/dataScienceSurveyBanner.ts (hunk 1 above)
Comment (IanMatthewHuff) [reply src/client/datascience/dataScienceSurveyBanner.ts:144]: Ahh, the shouldShow didn't need to be async. Yeah, that looks good now.

**Review by IanMatthewHuff — APPROVED** [review]

File: src/client/datascience/insidersNativeNotebookSurveyBanner.ts [hunk 3]
```diff
@@ -85,13 +85,17 @@ export class InsidersNativeNotebooksSurveyBanner implements IJupyterExtensionBan
...
     public async showBanner(): Promise<void> {
+        if (this.disabledInCurrentSession) {
+            return;
+        }
+        // Disable for the current session.
+        this.disabledInCurrentSession = true;
         const executionCount: number = this.getExecutionCount();
         const notebookCount: number = this.getOpenNotebookCount();
-        const show = await this.shouldShowBanner(executionCount, notebookCount);
+        const show = this.shouldShowBanner(executionCount, notebookCount);
```
Comment (rchiodo) [inline src/client/datascience/insidersNativeNotebookSurveyBanner.ts:95]: I must be missing something. showShowBanner is still async.

**Review by rchiodo — CHANGES_REQUESTED** [review]
⏲️

File: src/client/datascience/insidersNativeNotebookSurveyBanner.ts (hunk 3 above)
Comment (DonJayamanne) [reply src/client/datascience/insidersNativeNotebookSurveyBanner.ts]: Damn.

**Review by rchiodo — APPROVED** [review]

**Pull request merged**